common/lib/common-definitions/src/logger.ts

```diff
@@ -120,3 +120,23 @@ export interface ITelemetryLogger extends ITelemetryBaseLogger {
      */
     shipAssert(condition: boolean, event?: ITelemetryErrorEvent): void;
 }
+
+// ////////////////////// fluidError.ts in common-definitions /////////////////////////////
+/**
+ * Can be decl merged to add stuff like packagename, odspErrorResponse, etc
+ * The logger impl can choose to log anything in here as appropriate, but by default it's not
+ */
+export interface ISensitiveDebugData {
+    innerError?: any;
+}
+
+export interface IFluidError {
+    errorType: string;
+    message: string;
+    getFluidTelemetryProps: () => ITelemetryProperties;  //* use property getters?
+    getSensitiveDebugData: () => ISensitiveDebugData & { stack: string };
```

Comment:
i don't get the `{ stack: string }` won;t that be on the innerError?

Reply:
There may be cases where we don't even want to log our own callstacks - as crippling as that sounds debugging-wise. e.g. if some 3P code is loaded that exposes who the tenant is, for instance (I don't believe this applies yet but is conceivable). @AndreiZe (I think?) has raised concerns about someone downloading our JS and trying to run it from their machine, in which case callstacks would contain their machine name.

I'm proposing that we leave it to the host (whoever implements the `ITelemetryBaseLogger`) to decide if/when callstack is logged.  I will also lobby hard to ensure that for the foreseeable future, Office keeps logging it :)

As for `innerError`, that will only be present if we are wrapping some foreign error.  And it might not be a proper `Error` object, it's just `any`.